plugins/source/plugin.go

```diff
@@ -5,14 +5,16 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/cloudquery/plugin-sdk/backend"
 	"github.com/cloudquery/plugin-sdk/caser"
+	"github.com/cloudquery/plugin-sdk/internal/backends/local"
 	"github.com/cloudquery/plugin-sdk/schema"
 	"github.com/cloudquery/plugin-sdk/specs"
 	"github.com/rs/zerolog"
 	"golang.org/x/sync/semaphore"
 )
 
-type NewExecutionClientFunc func(context.Context, zerolog.Logger, specs.Source) (schema.ClientMeta, error)
+type NewExecutionClientFunc func(context.Context, zerolog.Logger, specs.Source, backend.Backend) (schema.ClientMeta, error)
```

Comment:
Let's introduce an optional struct here so we can have a way of both adding required parameters that will break the API and once that are not required like this one (i.e not all plugins have to implement/use cursor)

Reply:
Looking into this now